DOCSP-48422: search index type (#170)
mongodb/docs-rust: source/fundamentals/indexes/atlas-search-indexes.txt

Overview
--------

In this guide, you can learn how to create and manage **Atlas Search and
Vector Search indexes**. These indexes allow you to use the following
features:

- :atlas:`Atlas Search </atlas-search/>`: Perform fast, full-text
  searches

- :ref:`Atlas Vector Search <rust-vector-search>`: Perform
  semantic (similarity) searches on vector embeddings

> Atlas Search and Vector Search indexes specify which fields to index,
> how these fields are indexed, and can set other optional configurations.

Reviewer: to keep a parallel structure, either the third item should inherit the verb 'specify', like the second item does, or the second item should supply its own verb

```suggestion
Atlas Search and Vector Search indexes specify which fields to index,
specify how these fields are indexed, and can set other optional configurations.
```

- :ref:`rust-update-search-index`
- :ref:`rust-drop-search-index`

.. note:: Sample Data

>    The examples in this guide access the ``embedded_movies`` collection

Reviewer: S: i think we say 'use' elsewhere. feels simpler, but not a big deal

```suggestion
   The examples in this guide use the ``embedded_movies`` collection
```

The {+driver-short+} provides the following ``SearchIndexModel`` builder methods:

.. list-table::
   :header-rows: 1
   :widths: 25 25 50

   * - Builder Method
     - Parameter Type
     - Description

   * - ``definition()``
     - ``Document``

>      - Specifies the index definition. If you exclude this setting, the

Reviewer: s:

```suggestion
     - Specifies the index definition. If you omit this setting, the
```

applies throughout

       driver creates an Atlas Search index with dynamic mappings.

   * - ``name()``
     - ``String``
     - Sets the index name. If you exclude this setting, the
       driver sets the name to ``default``.

   * - ``index_type()``
     - ``SearchIndexType``
     - Sets the index type. If you exclude this setting, the
       driver creates an Atlas Search index by default.

To learn more about Atlas Search field mappings, see :atlas:`Define Field Mappings
</atlas-search/define-field-mappings/>` in the Atlas documentation.

To learn more about defining Atlas Vector Search indexes, see
:atlas:`How to Index Fields for Vector Search
</atlas-vector-search/vector-search-type/>` in the Atlas documentation.

Example Models
~~~~~~~~~~~~~~

> The following example creates specifications for a index named ``search_idx`` in a

Reviewer: ```suggestion
The following example creates specifications for an index named ``search_idx`` in a
```

> The following example creates specifications for a index named ``search_idx`` in a
> ``SearchIndexModel`` instance. The code sets a static mapping to index only the ``title``

Reviewer: s: maybe ambiguous about the index being created in the instance instead of via it

```suggestion
The following example uses a ``SearchIndexModel`` instance to create
specifications for a index named ``search_idx``. The code sets a static
mapping to index only the ``title``
```

and ``released`` fields:

.. literalinclude:: /includes/fundamentals/code-snippets/indexes/atlas-indexes.rs
   :start-after: begin-as-model
   :end-before: end-as-model
   :language: rust
   :dedent:

> The following example creates specifications for a index named ``vs_idx`` in a

Reviewer: ```suggestion
The following example creates specifications for an index named ``vs_idx`` in a
```

> The following example creates specifications for a index named ``vs_idx`` in a
> ``SearchIndexModel`` instance. The code specifies the embedding path as

Reviewer: ```suggestion
The following example uses a ``SearchIndexModel`` instance to create specifications
for a index named ``vs_idx``. The code specifies the embedding path as
```

Example Models
~~~~~~~~~~~~~~

The following example creates a ``SearchIndexModel`` instance to provide

> specifications for a index named ``search_idx``. The code specifies static

Reviewer: ```suggestion
specifications for an index named ``search_idx``. The code specifies static
```

mappings of the ``title`` and ``released`` fields:

.. literalinclude:: /includes/fundamentals/code-snippets/indexes/atlas-indexes.rs
   :start-after: begin-as-model
   :end-before: end-as-model
   :language: rust
   :dedent:

The following example creates a ``SearchIndexModel`` instance to provide

> specifications for a index named ``vs_idx``. The code specifies the

Reviewer: ```suggestion
specifications for an index named ``vs_idx``. The code specifies the
```

Create Multiple Search Indexes
------------------------------

You can create multiple Atlas Search and Vector Search indexes
by calling the ``create_search_indexes()`` method on a ``Collection``
instance. This method accepts a vector of ``SearchIndexModel`` instances
as a parameter.

Example
~~~~~~~

The following example creates an Atlas Search and a Vector Search index

> on the ``embedded_movies`` collection. The code creates ``SearchIndexModel``

Reviewer: S: clarify one instance per index and remove ambiguity

(suggestion spans the next line too): "For each index, the code creates a ``SearchIndexModel`` instance that specifies the index name and definition."

Author: changing to a list of tasks

instances for each index that specify the index names and definitions. Then, the code
passes these models as a vector to the ``create_search_indexes()``
method to create the indexes.

.. io-code-block::
   :copyable: true

   .. input:: /includes/fundamentals/code-snippets/indexes/atlas-indexes.rs
      :start-after: begin-atlas-create-many
      :end-before: end-atlas-create-many
      :language: rust
      :emphasize-lines: 20
      :dedent:

   .. output::
      :language: console
      :visible: false

      Created Atlas Search indexes:
      ["as_idx", "vs_idx"]

.. _rust-list-search-index:

List Search Indexes
-------------------

You can access information about a collection's existing Atlas Search
and Vector Search indexes by calling the ``list_search_indexes()``
method on the collection.

Example
~~~~~~~

The following example accesses information about the Atlas Search and
Vector Search indexes created in the :ref:`rust-create-search-indexes`
section of this page. The code calls the ``list_search_indexes()``

> method and outputs all the Search indexes on the collection:

Reviewer: Q: i assume this means 'all the Atlas/Vector Search indexes'. if so, probably worth specifying since there are other types of indexes that are used for searching, and i don't think enough people will catch the use of capital S in Search
